### PR TITLE
chapter 12 : Reset password

### PR DESCRIPTION
**Pull request by DaudKhadijah01**

### CODE REVIEW

**Comment by DaudKhadijah01** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 class User < ApplicationRecord
+<<<<<<< HEAD
   attr_accessor :remember_token, :activation_token
+=======
+  attr_accessor :remember_token, :activation_token, :reset_token
+>>>>>>> chapter_9
```
Comment (ghost) [inline app/models/user.rb]: fix conflict code please

File: app/views/user_mailer/password_reset.html.erb [hunk 2]
```diff
@@ -1,5 +1,10 @@
-<h1>User#password_reset</h1>
+<h1>Password reset</h1>
 
-<p>
-  <%= @greeting %>, find me in app/views/user_mailer/password_reset.html.erb
+<p>To reset your password click the link below:</p>
+
+  <%= link_to "Reset password", edit_password_reset_url(id: @user.reset_token, email: @user.email) %>
+
+<p>This link will expire in two hours.</p>
```
Comment (ghost) [inline app/views/user_mailer/password_reset.html.erb]:
```suggestion
This link will expire in two hours.
```

File: app/views/user_mailer/password_reset.html.erb [hunk 3]
```diff
@@ -1,5 +1,10 @@
-<h1>User#password_reset</h1>
+<h1>Password reset</h1>
 
-<p>
-  <%= @greeting %>, find me in app/views/user_mailer/password_reset.html.erb
+<p>To reset your password click the link below:</p>
```
Comment (ghost) [inline app/views/user_mailer/password_reset.html.erb]:
```suggestion
<p>To reset your password click the link below:
```

File: config/routes.rb [hunk 4]
```diff
@@ -16,5 +16,6 @@
     delete "/logout", to: "sessions#destroy"
 
     resources :account_activations, only: [:edit]
+    resources :password_resets, only: [:new, :create, :edit, :update]
```
Comment (ghost) [inline config/routes.rb]: use except please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by DaudKhadijah01** [comment]
ready

File: reset_sent_at:datetime [hunk 5]
```diff
@@ -0,0 +1,2 @@
+      invoke  active_record
```
Comment (ngocvt-0484) [inline reset_sent_at:datetime]: remove this file please

File: app/models/user.rb [hunk 6]
```diff
@@ -50,6 +50,22 @@ def send_activation_email
     UserMailer.account_activation(self).deliver_now
   end
 
+  # Sets the password reset attributes.
+  def create_reset_digest
+    self.reset_token = User.new_token
+    update_attribute(:reset_digest, User.digest(reset_token))
+    update_attribute(:reset_sent_at, Time.zone.now)
```
Comment (ngocvt-0484) [inline app/models/user.rb]: you can use update_attributes 
```suggestion
    update_attributes(reset_sent_at: Time.zone.now, reset_digest: User.digest(reset_token))
```

**Comment by DaudKhadijah01** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**